gfsphysics/GFS_layer/GFS_typedefs.meta

```diff
@@ -7661,6 +7661,13 @@
   dimensions = (horizontal_dimension,vertical_dimension)
   type = real
   kind = kind_phys
+[save_tcp]
+  standard_name = air_temperature_save_from_cumulus_paramterization
```

Comment:
@grantfirl do you have any input on the choice of the standard name?

Reply:
Hmm. I don't immediately see why this is needed in addition to "air_temperature_save" that already exists. If it is indeed necessary, I thought we had been using "scheme" instead of parameterization and "deep_convection" or "shallow_convection", specifically, instead of "cumulus".

Reply:
I think Hannah's approach is correct. save_tcp is storing the value before entering the deep convection scheme, and it is used after returning from the shallow convection scheme to approximate the cloud ice number concentration change due to deep+shallow convection.

If we were using save_t (air_temperature_save) and ldiag3d was true, then we would get the temperature after deep but before shallow convection, not the temperature before shallow convection.

However, there are problems with this approach: if no deep or shallow convection are used, the DCNV and SCNV generic schemes may not be run (see, for example, suite_FV3_GSD_SAR.xml - but I am not sure if this is correct) and save_tcp not set correctly. To me, this raises the question whether the new code in lines 749-773 in GFS_suite_interstitial_4 should only be executed if Thompson AND (deep OR shallow convection) are used?

Given that the calculation currently sits in GFS_suite_interstitial_4, one could move the saving of the temperature before convection into GFS_suite_interstitial_3. Or, thinking further, having two interstitial schemes before GFS_DCNV_generic_pre and after GFS_SCNV_generic_post that capture logic for deep+shallow convection?

@hannahcbarnes @grantfirl what do you think?

Reply:
Dom is correct, I made save_tcp since save_t stores the temperature well before the convection is called so temperature - save_t contains temperature changes that come from more than just the convective schemes. 

I understand the problem of save_tcp when shallow and/or deep are not called. I agree with Dom's idea of adding an if statement around the code in lines 749-773. I think it should be if Thompson and (deep or shallow) convection. 

I think it would be best to move the saving of the temperature into GFS_suite_interstitial_3. I think that is actually better than hacing in in GFS_DCNV_pre since it is right before the convective schemes are called. 

In terms of where to put the calculation that is currently in lines 749-773. The current location certainly does lump the shallow and deep convection together. However, I ran a few tests to see if separating the deep from the shallow and I did not see much of a difference. If we want to separate the impact of deep and shallow schemes, we can defintely add those two intersitial schemes. However, I am not sure that is necessary. 

@DomHeinzeller do you agree?

Reply:
I think that GFS_suite_interstitial_3 and GFS_suite_interstitial_4 are the best places, given the re-arrangement in the SDF.

Reply:
I agree. Do you want me to make save_tcp into GFS_suite_interstitial_3 or will you make that change when you do the merger?

Reply:
I'll be doing this as part of the update to the latest code base, no worries. Will ask you to thoroughly review it - hopefully tomorrow.